Realm/RLMQueryUtil.mm

```diff
@@ -1312,6 +1081,8 @@ void convert_null(T&& value, Fn&& fn) {
 
 template<typename T>
 using Identity = T;
+template<typename>
+using AlwaysDictionary = Dictionary;
```

Comment:
It looks like this is only used in one spot, what is the benefit of using this approach?

Reply:
`do_add_constraint` takes a template template parameter which is used to map a type `T` into `Columns<Lst<T>>`, `Columns<Set<T>>`, or `Columns<T>`. This lets us use the same code path for Dictionary as well rather than duplicating it. Unlike the other collections Dictionary doesn't have a template parameter, so the dictionary template template ignores its template parameter and always produces `Columns<Dictionary>` regardless of `T`. Note that `T` is still used for the other parameter in the expression being built, so we can't skip the switch on type for Dictionaries even though the column is always resolved to the same type.